### PR TITLE
new `to_lower_camel()` function

### LINKED ISSUE #3464: pydantic.utils.to_camel() is actually to_pascal()
### Checks

* [ y ] I added a descriptive title to this issue
* [ y ] I have searched (google, github) for similar issues and couldn't find anything
* [ y ] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.8.1
            pydantic compiled: True
                 install path: /home/schlerp/projects/pelt-studio/venv/lib/python3.8/site-packages/pydantic
               python version: 3.8.5 (default, Sep  4 2020, 07:30:14)  [GCC 7.3.0]
                     platform: Linux-5.15.2-zen1-1-zen-x86_64-with-glibc2.10
     optional deps. installed: ['typing-extensions']
```

Camel case and pascal case are similar however, they differ by the capitalisation of the first letter. The current implementation of camel_case in pydantic us actually pascal case and not camel case at all. I suggest renaming this and also implementing a camel case. See below for code expressing the issue and suggested fix.

**Pascal Case** (aka init case or upper camel case)
All spaces/underscores removed and the start of every word is capitalised.

**Camel Case** (aka lower camel case)
All spaces and underscores removed and the start of every word, is capitalised, except the first word which is always lower case.

Issue:
```py
from pydantic.utils import to_camel

valid_pascal = "PascalCase"
valid_camel = "camelCase"

example = to_camel("i_shouldnt_be_capitalised")

assert valid_pascal == to_camel("pascal_case")
assert valid_camel != to_camel("camel_case")
```

suggested fix, rename `to_camel()` -> `to_pascal()`, and write new `to_camel()` function:

```py
def to_pascal(string: str) -> str:
    return "".join(word.capitalize() for word in string.split("_"))


def to_camel(string: str) -> str:
    if len(string) >= 1:
        pascal_string = to_pascal(string)
        return pascal_string[0].lower() + pascal_string[1:]
    return string.lower()

```

Alternatively, if there is code which will break because it is dependent on the `camel_case()` function remaining pascal case, then i propose we implement a new function called `to_lower_camel()` which implements the first letter lower case variant:

```py
def to_camel(string: str) -> str:
    return "".join(word.capitalize() for word in string.split("_"))


def to_lower_camel(string: str) -> str:
    if len(string) >= 1:
        pascal_string = to_camel(string)
        return pascal_string[0].lower() + pascal_string[1:]
    return string.lower()

```

### PR DESCRIPTION
**Pull request by schlerp**
I needed non pascal case camel case support

## Change Summary

created new function (`to_lower_camel()` for non pascal case camel case), unit tests, and updated documentation:

```py
>>> to_camel("i_like_turtles")
ILikeTurtles

>>> to_lower_camel("i_like_turtles")
iLikeTurtles

```

## Related issue number

fix #3464

## Checklist

* [ y ] Unit tests for the changes exist
* [ y ] Tests pass on CI and coverage remains at **previous level**
* [ y ] Documentation reflects the changes where applicable
* [ y ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ y ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by schlerp** [comment]
please review

**Comment by schlerp** [comment]
apologies about the number of commits, am working from my phone while in transit and I cant work out how to batch commits from the web ui (or to make a fork with out opening a merge straight away **facepalm**)

**Review by PrettyWood — COMMENTED** [review]
I think renaming `to_camel` to `to_pascal` and having a proper `to_camel` is better.
We can add a breaking change and people who rely on old `to_camel` will just have to replace everywhere with `to_pascal`, which is trivial to make

**Comment by schlerp** [comment]
@PrettyWood thank s for taking a look. I felt it was a nicer solution aswell and if we are allowed to make a change like that then I say we go for it. 

Sorry first time submitting a PR for pydantic: 
Is your approval enough? should I alter this PR to reflect our conversation or is there quorum we need to meet first?

**Comment by samuelcolvin** [comment]
I agree with @PrettyWood, on naming

Couple of things that might be useful for you to know in future:
* please put "fixes #3464" in the PR body, so the issue is automatically closed when this PR is merged, just referencing the issue number is useful, but not sufficient to auto-close it
* you can use specific syntax for checklists in github: `[x]` or `[ ]` and they'll come through as proper checkboxes

**Comment by samuelcolvin** [comment]
please update

**Comment by samuelcolvin** [comment]
Given that this is for v1.10, let's not change the name. I'll fix this up.

**Comment by samuelcolvin** [comment]
thanks so much.

File: tests/test_utils.py [hunk 1]
```diff
@@ -528,6 +529,18 @@ def test_undefined_pickle():
...
+
+
+def test_on_lower_camel_one_length():
+    assert to_lower_camel('a') == 'a'
+
+
+def test_on_lower_camel_many_length():
+    assert to_lower_camel('i_like_turtles') == 'iLikeTurtles'
+
+
```
Comment (satheler) [inline tests/test_utils.py:543]: could be written as follows

```suggestion
@pytest.mark.parametrize(
    'value, expected_result',
    [('', ''), ('a', 'a'), ('i_like_turtles', 'iLikeTurtles')],
)
def test_on_lower_camel(value: str, expected_result: str):
  assert to_lower_camel(value) == expected_result
```

**Pull request merged**